### ISSUE TITLE
Move elasticsearch `endpoint` and `region` options under `aws` namespace/table

### ISSUE THREAD

**Issue by binarylogic** [timberio/vector#1608]
After auditing some of the elasticsearch docs I found the `region` and `endpoint` options to be very confusing. I'm surprised that we added all of these options without noticing the confusion or adding docs to explain how they work.

From what I understand:

1. The `endpoint` option is accepted but only used for authentication?
2. The `region` option is accepted but only used for authentication?

My questions are:

1. How exactly do these optionss work?
2. Under what circumstances are these used for authentication? 
3. Would it make sense to move these options under an `aws` table?

**Comment by binarylogic** [comment]
@bruceg are you the best person to answer this or @LucioFranco?

**Comment by Jeffail** [comment]
These are AWS specific fields so not necessarily the same category as authentication.

With AWS specific components it makes sense for these fields to be at the root of the component config, but in the case of ES it _could_ be targeting Amazon Elastic Service or just a regular ES cluster. Having them at the root makes the configuration more consistent with dedicated AWS components, but in the latter case these fields do nothing and might be confusing for people with no AWS experience.

We could either add more clarification to the field docs or, in the case where a component is _capable_ of connecting to AWS APIs but also has a clear application outside of the AWS ecosystem, put AWS fields into their own namespace (`aws`).

**Comment by binarylogic** [comment]
Yeah, I'm in favor of nesting them under `aws` if that makes the most sense. It's not clear to me how they're used though.

**Comment by lukesteensen** [comment]
When you're using AWS's ES service, you can specify either the `region` (which we can turn into a URL/endpoint internally) or an endpoint for us to use directly. They should be mutually exclusive, used only with AWS, and serve as the indicator that you're using AWS and we should use AWS-style authentication.

**Comment by bruceg** [comment]
@lukesteensen is correct.

1. AWS segments their services into regions, and a different hostname is used to access each region. These two options allow for choosing either a predefined region to connect to, or a literal endpoint URL (which we use for testing with the `localstack` container).

2. These are not directly used for authentication, except for the `elasticsearch` sink that can connect either directly to an ElasticSearch server or to AWS' ES service, which changes which authentication mechanism needs to be used.

3. Since they are mutually exclusive, I'm not sure nesting them makes much sense, except maybe for `elasticsearch` to indicate this is specifically using the AWS mode of connection.

**Comment by binarylogic** [comment]
I think moving them under `aws` specifically for the `elasticsearch` sink makes sense. Especially if we plan to support other Elasticsearch providers:

```toml
[sinks.es]
  type = "elasticsearch"

  [sinks.es.aws]
    endpoint = "..."
    region = "..."
```

If you agree, let's make this change since it's somewhat confusing to have both an `endpoint` and `host` top-level options outside of the context of AWS.

**Comment by binarylogic** [comment]
I'm also open to removing the `endpoint` and `region` options in favor of the `host` option, but I thought this tied into authentication somehow?

**Comment by binarylogic** [comment]
Let me know if you agree @bruceg, I'm re-assigning based on my comments.

**Comment by bruceg** [comment]
I agree with the `elasticsearch` sink change, and will do that.

**Comment by bruceg** [comment]
Closed in favor of the change in #1703
